sdformat_urdf/CMakeLists.txt

```
@@ -0,0 +1,175 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(sdformat_urdf)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Wnon-virtual-dtor -Woverloaded-virtual)
+endif()
+
+find_package(ament_cmake_ros REQUIRED)
+
+find_package(pluginlib REQUIRED)
+find_package(rcutils REQUIRED)
+find_package(sdformat9 REQUIRED)
+find_package(urdfdom_headers REQUIRED)
+find_package(urdf_parser_plugin REQUIRED)
+find_package(tinyxml2_vendor REQUIRED)
+find_package(TinyXML2 REQUIRED)
+
+# Add sdformat_urdf shared library
+add_library(sdformat_urdf SHARED
+  src/sdformat_urdf.cpp
+)
+target_link_libraries(sdformat_urdf
+  PUBLIC
+    sdformat9::sdformat9
+)
+target_include_directories(sdformat_urdf
+  PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>"
+    ${urdfdom_headers_INCLUDE_DIRS}
+)
+ament_target_dependencies(sdformat_urdf PUBLIC
+  "rcutils"
+)
+
+# Add sdformat_urdf_plugin module library
+add_library(sdformat_urdf_plugin MODULE
+  src/sdformat_urdf_plugin.cpp
+)
+target_link_libraries(sdformat_urdf_plugin PUBLIC
+  sdformat_urdf
+)
+ament_target_dependencies(sdformat_urdf_plugin PUBLIC
+  "pluginlib"
+  "rcutils"
+  "TinyXML2"
+  "urdf_parser_plugin"
+)
+
+ament_export_dependencies(urdfdom_headers)
+ament_export_dependencies(sdformat9)
+
```

I tried using `sdformat_urdf` in another library and it failed saying 
```
Target "test_using_sdformat_urdf" links to target "rcutils::rcutils" but the target was not found.  Perhaps a find_package() call is missing for an IMPORTED target, or an ALIAS target is missing?
```
I think since you have `ament_target_dependencies(sdformat_urdf PUBLIC "rcutils")` [here](https://github.com/ros/sdformat_urdf/pull/1/files#diff-691bc8ec14d67d8f2a9c492b7b2b9e947b0dc037a3fbb6b6df4b5916c80ac47fR37), we should add 
```suggestion
ament_export_dependencies(rcutils)
```
Alternatively, we could make `rcutils` a private dependency, but I'm not sure we can do that using `ament_target_dependencies`.

Here the CMakeLists.txt I used.

```cmake
cmake_minimum_required(VERSION 3.5)
project(test_using_sdformat_urdf)

find_package(sdformat_urdf REQUIRED)

add_executable(test_using_sdformat_urdf src/test_using_sdformat_urdf.cpp)
target_link_libraries(test_using_sdformat_urdf PUBLIC sdformat_urdf::sdformat_urdf)

install(TARGETS test_using_sdformat_urdf RUNTIME DESTINATION bin)
```

---

> Alternatively, we could make rcutils a private dependency, but I'm not sure we can do that using ament_target_dependencies.

I replaced `ament_target_dependencies()` with `target_link_libraries()` in f25ffe2

ament/ament_cmake#292